src/include/miopen/generic_search.hpp

```diff
@@ -330,12 +331,31 @@ void CompileAgent(size_t thread_index,
         if(current_time - start_time > time_budget)
         {
             MIOPEN_LOG_I2("Thread: " << thread_index << " Done, exhausted time budget");
-            auto tmp = std::make_tuple<PerformanceConfig, ConvSolution, bool>({}, {}, true);
-            comp_queue.push(std::move(tmp));
             break;
         }
         auto& current_config          = data.at(idx);
         ConvSolution current_solution = s.GetSolution(context, problem, current_config);
+
+        // Get provided workspace size or treat as unlimited if not implemented
+        std::size_t provided_size = std::numeric_limits<std::size_t>::max();
+
+        try
+        {
+            provided_size = invoke_ctx.GetWorkspaceSize();
+        }
+        catch(const miopen::Exception&)
+        {
+            MIOPEN_LOG_I2("CompileAgent: Workspace size not implemented, treating as unlimited.");
+        }
+
+        if(provided_size < current_solution.workspace_sz)
+        {
+            MIOPEN_LOG_I2("Thread: " << thread_index
+                                     << " Skipping config due to workspace requirement "
+                                     << current_solution.workspace_sz << " > " << provided_size);
+            continue;
+        }
```

Comment:
I like this check. Though it seems like generic search is insisting that a workspace size be constant across as solver: https://github.com/ROCm/MIOpen/blob/afb4feaac60af16f783140764eb671fb6e05d0a7/src/include/miopen/generic_search.hpp#L477-L484

Reply:
Not sure if I got the question right, with provided workspace size set to 1073741824,  both default_solution.workspace_sz and current_solution.workspace_sz are 5452602112. Their values are different from user provided workspace size

MIOpen(HIP): Error [GenericSearch] ********default_solution.workspace_sz: 5452602112
MIOpen(HIP): Error [GenericSearch] ********current_solution.workspace_sz: 5452602112

Below is the complete log (without my changes)
[solution.workspace_sz.txt](https://github.com/user-attachments/files/20349629/solution.workspace_sz.txt)